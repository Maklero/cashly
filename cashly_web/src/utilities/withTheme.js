import { ThemeProvider } from "styled-components";
import { theme } from "../styles";

const withTheme = (Component) => {
    return (
        <ThemeProvider theme={theme}>
            <Component />
        </ThemeProvider>
    );
}

export default withTheme;