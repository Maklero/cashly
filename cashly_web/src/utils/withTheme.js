import { ThemeProvider } from 'styled-components'

import { theme } from '@/styles'

export const withTheme = (component) => (
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
)
