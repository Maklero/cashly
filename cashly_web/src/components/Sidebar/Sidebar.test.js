import { render } from "@testing-library/react";
import { Sidebar } from "..";
import withTheme from "../../utilities/withTheme";

test("renders sidebar component", () => {
    const { container } =  render(withTheme(Sidebar));

    expect(container.firstChild).toMatchSnapshot();
})