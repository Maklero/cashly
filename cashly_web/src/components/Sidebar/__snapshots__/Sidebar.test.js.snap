// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sidebar component 1`] = `
<nav
  class="sc-hKwDye bcwUzE"
>
  <button
    class="sc-dkPtRN VNSFh"
  >
    Wydatki
  </button>
  <button
    class="sc-dkPtRN VNSFh"
  >
    Kategorie wydatk√≥w
  </button>
</nav>
`;
